src/api/services/script.js

```diff
@@ -0,0 +1,58 @@
+import Path from 'path';
+import fs from 'fs';
+import ezs from 'ezs';
+import URL from 'url';
+import fetch from 'fetch-with-proxy';
+import config from '../../../config.json';
+
+export default class Script {
+    constructor(source) {
+        const routineDirectory = `./${source}/`;
+        const routineLocalDirectory = Path.resolve(
+            __dirname,
+            '../',
+            routineDirectory,
+        );
+        const routinesLocal = config.routines
+            .map(routineName =>
+                Path.resolve(routineLocalDirectory, routineName.concat('.ini')),
+            )
+            .filter(fileName => fs.existsSync(fileName))
+            .map(fileName => [
+                fileName,
+                ezs.metaFile(fileName),
+                Path.basename(fileName, '.ini'),
+                fs.readFileSync(fileName).toString(),
+            ]);
+        const pluginsURL = config.pluginsURL || '';
+        const routineRepository = URL.resolve(pluginsURL, routineDirectory);
+        const routinesDistant = config.routines
+            .map(routineName =>
+                URL.resolve(routineRepository, routineName.concat('.ini')),
+            )
+            .map(fileName => [
+                fileName,
+                null,
+                Path.basename(fileName, '.ini'),
+                null,
+            ]);
+        this.local = routinesLocal;
+        this.distant = routinesDistant;
+    }
+
+    async get(routineCalled) {
+        const routineLocal = this.local.find(r => r[2] === routineCalled);
+        const routineDistant = this.distant.find(r => r[2] === routineCalled);
+        if (!routineLocal && routineDistant) {
+            const response = await fetch(routineDistant[0]);
+            const routineScript = await response.text();
+            if (routineScript) {
+                routineDistant[1] = ezs.metaString(routineScript);
+                routineDistant[3] = routineScript;
+            }
+        }
+        const routine = routineLocal || routineDistant;
```

Comment:
yes but we  can change... What is your preference ?

Reply:
I think that when we specify a new version of a routine, we expect it to prevail on the default one.
Thus, it would be more natural to use `routineDistant || routineLocal`.